Working minimal ScoringPlugin example

j-herman opened the pull request:
This PR does not need to be merged - if you prefer, we can leave the branch as-is for reference.

This code implements a very basic version of the ScoringPlugin.  It loads the plugin, keeps track of the task time, and publishes changes in state (initial/ready/running/finished) to the /vrx/task/info topic.  Timing and some of messaging is stolen from Rishikesh's in-progress branch.  The goal is to provide some capability for testing new code segments in plugins that will derive from the base ScoringPlugin while full functionality is pending.

To test: start the simulation via `ros2 launch vrx_ros competition_local.launch.py ign_args:="-v 4 -r sydney_regatta.sdf"`
You should be able to see the ScoringPlugin listed under System Plugin info in the Component Inspector.
In a separate terminal window, echo the task topic via `ign topic --echo -t /vrx/task/info`
You should see the state message updating at 1Hz.  It will cycle through from initial to ready to running.  Finally, it will print "finished" once, and the simulation will exit.

Next steps: Ran into significant cmake problems trying to incorporate protobuf custom messages, and so the task message in this example is minimal.  Next major effort will be to resolve that issue.

M1chaelM commented:
This runs fine for me and I see the behavior described. We should discuss whether to use this or PR #486 as a basis for moving forward.

j-herman commented:
Made some modifications to better support inheritance based on work on the Stationkeeping scoring plugin - need to build this library first.  Changed a few function declarations that shouldn't be required overrides, and fixed initialization of timing variables.

caguero commented:
Replaced with #501.
